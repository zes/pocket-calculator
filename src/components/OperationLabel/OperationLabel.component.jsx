import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

export default class OperationLabel extends PureComponent {
  render() {
    const { operation } = this.props;

    return <span>{operation}</span>;
  }
}

OperationLabel.propTypes = {
  operation: PropTypes.string.isRequired,
};
