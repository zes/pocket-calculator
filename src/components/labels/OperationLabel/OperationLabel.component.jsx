import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import GenericLabel from '../GenericLabel/GenericLabel.component';

export default class OperationLabel extends PureComponent {
  render() {
    const { operation } = this.props;

    return <GenericLabel className="operation" value={operation} />;
  }
}

OperationLabel.propTypes = {
  operation: PropTypes.string.isRequired,
};
