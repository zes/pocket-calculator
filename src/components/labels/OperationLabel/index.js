export { default } from './OperationLabel.container';
