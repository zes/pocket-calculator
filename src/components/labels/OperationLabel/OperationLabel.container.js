import { connect } from 'react-redux';

import OperationLabel from './OperationLabel.component';

const mapStateToProps = state => ({
  operation: state.operation,
});

export default connect(mapStateToProps)(OperationLabel);
