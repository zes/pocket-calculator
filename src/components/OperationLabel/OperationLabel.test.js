import React from 'react';
import renderer from 'react-test-renderer';
import OperationLabel from './OperationLabel.component';

describe('[Component] <OperationLabel />', () => {
  const props = {
    operation: '2+4/78.9',
  };

  it('renders a snapshot', () => {
    const tree = renderer.create(<OperationLabel {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
