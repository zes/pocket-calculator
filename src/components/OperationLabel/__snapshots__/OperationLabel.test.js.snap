// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] <OperationLabel /> renders a snapshot 1`] = `
<span>
  2+4/78.9
</span>
`;
